includes/embeds/class-amp-gutenberg-categories-handler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @since 1.0
+ */
+class AMP_Gutenberg_Categories_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Register embed.
+	 */
+	public function register_embed() {
+		if ( function_exists( 'gutenberg_init' ) ) {
+			add_action( 'the_post', array( $this, 'override_category_block_render_callback' ) );
+		}
+	}
+
+	/**
+	 * Unregister embed.
+	 */
+	public function unregister_embed() {}
+
+	/**
+	 * Override the output of Gutenberg core/category block.
+	 */
+	public function override_category_block_render_callback() {
+		if ( is_amp_endpoint() ) {
```

Comment:
Note that `register_embed` will have only been called if it is an AMP endpoint to begin with. So this `override_category_block_render_callback` method can actually be eliminated in favor of moving the logic to `register_embed` directly, minus the `is_amp_endpoint` check.